Kiss.Bff/Extern/Elasticsearch/ElasticsearchController.cs

```diff
@@ -0,0 +1,153 @@
+        private readonly ILogger<ElasticsearchController> _logger;
+
+        public ElasticsearchController(
+            IHttpClientFactory httpClientFactory,
+            IConfiguration configuration,
+            ILogger<ElasticsearchController> logger)
+        {
+            _httpClientFactory = httpClientFactory;
+
+            // Load configuration
+            _elasticsearchBaseUrl = configuration["ELASTIC_BASE_URL"] ?? throw new InvalidOperationException("ELASTIC_BASE_URL not configured");
+            _elasticsearchUsername = configuration["ELASTIC_USERNAME"] ?? throw new InvalidOperationException("ELASTIC_USERNAME not configured");
+            _elasticsearchPassword = configuration["ELASTIC_PASSWORD"] ?? throw new InvalidOperationException("ELASTIC_PASSWORD not configured");
+            _logger = logger;
+        }
+
+        [HttpPost("{index}/_search")]
+        [Authorize(Policy = Policies.KcmOrKennisbankPolicy)]
+        public async Task<IActionResult> Search([FromRoute] string index)
+        {
+            var cancellationToken = Request.HttpContext.RequestAborted;
+
+            try
+            {
+                // Read and parse request body
+                string requestBody;
+                using (var reader = new StreamReader(Request.Body, Encoding.UTF8))
+                {
+                    requestBody = await reader.ReadToEndAsync();
+                }
+
+                if (string.IsNullOrWhiteSpace(requestBody))
+                {
+                    return BadRequest(new { error = "Request body cannot be empty" });
+                }
+
+                JsonObject? elasticqQuery;
+                try
+                {
+                    var queryNode = JsonNode.Parse(requestBody);
+                    elasticqQuery = queryNode?.AsObject();
+
+                    if (elasticqQuery == null)
+                    {
+                        return BadRequest(new { error = "Invalid query format: expected JSON object" });
+                    }
+                }
+                catch (JsonException ex)
+                {
+                    _logger.LogError(ex, "Invalid JSON in request body");
+                    return BadRequest(new { error = "Invalid JSON in request body" });
+                }
+
+                // Transform request (apply role-based filtering)
+                ApplyRequestTransform(elasticqQuery);
+
+                // Forward request to Elasticsearch
+                var modifiedRequestBody = elasticqQuery.ToJsonString();
+
+                var httpRequest = new HttpRequestMessage(HttpMethod.Post, $"{_elasticsearchBaseUrl}/{index}/_search")
+                {
+                    Content = new StringContent(modifiedRequestBody, Encoding.UTF8, "application/json")
+                };
+
+                ApplyAuthenticationHeaders(httpRequest.Headers);
+
+                // Send request
+                var client = _httpClientFactory.CreateClient();
```

Comment:
je maakte een specifieke eleastic httpclient aan, maar volgens mij gebruik je hier de default. Moet je hier dan niet dit doen? `CreateClient("elasticsearch")`
en zo ja dan is een typed client wellicht mooier: https://learn.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-8.0#typed-clients